### ISSUE TITLE
Mention username/ping user in commands such as !embed

### ISSUE THREAD

**Issue by laundmo** [python-discord/bot#1382]
Currently, commands such as `!embed` or `!rule` can seem untargeted when sent in channels. To solve this I propose mentioning the username or even pinging the user.

There are multiple ideas I have how we could go about this:
- make the user an optional argument. By far the most obvious and simple choice. `!paste 12345667` or `!paste @username` but the issue is that this could ping the user twice
- additionally or alternatively get the user from a reply, so you can reply with `!paste` (and turn off the ping)
- make it a slash command, since those won’t be sent to the channel, the duplicate pinging won't happen.

**Comment by mbaruh** [comment]
"commands such as !embed" do you mean paste?

**Comment by Akarys42** [comment]
Using something like `!paste @user` will work with the current setup. You can even use the new replies to add context to your tag. I’m not sure if having another ping from the bot which would be just under the command invocation is a good idea.

**Comment by laundmo** [comment]
@Akarys42 while its trues that you can ping in the command, I still see people ignore and be confused who the bot command was meant for. But I agree, double pings should definitely be considered.

**Comment by gustavwilliam** [comment]
I'm not sure that making the bot ping the user will do much to help. If they ignored the ping with the command invocation, it seems unlikely that they'll care about the bot embed.

There's also the previously mentioned issue with double pings, which we should definitely avoid. Using an ID seems too complex, and I believe  most people would still ping them. This could be solved by only including a ping in the bot message if the invocation included an ID, and not if the user was mentioned, but that seems like a pretty inconsistent implementation.

All in all, I like the ambition, but I don't think this would make sense to implement like what was proposed.

**Comment by laundmo** [comment]
i have thought about this some more, and the best we can do would be to put their username into the embed, to maie it seem more targeted. Also, i know from experience that a username in abot messafe can catch your eye way more than a unpersonalised one. The main issue to discuss then is if and how these could be inserted in the text, and what to do if no user is passed.

**Comment by gustavwilliam** [comment]
Putting the username (and avatar) in bot embeds in commonly used to signal who invoked the command, on other bots. Seeing my own name and possibly avatar in something like a resources embed because someone pinged me doesn't seem very appealing, and might make me think that I somehow invoked the command. I'm not a huge fan of the idea.

**Comment by laundmo** [comment]
@gustavwilliam i agree for the embedding of username/avatar like other bots do, i was more thinking of putting the username in the actual text along the lines of 

```
Hey {username}, if you want to paste code you can use paste.pydis.com
```

this would not seem like the mentioned person invoked the command

**Comment by MrKomodoDragon** [comment]
I like this idea!

**Comment by gustavwilliam** [comment]
That *might* work, but tags like `!paste` are in embeds. How would this be combined with the idea you just proposed?

**Comment by ChrisLovering** [comment]
You can mention users in embed bodies, they just won't get notified. 
![image](https://user-images.githubusercontent.com/9753350/107159533-96b75c80-6988-11eb-95fb-257f089d024e.png)

**Comment by MrKomodoDragon** [comment]
Good Idea Chris!

**Comment by MarkKoz** [comment]
Since it's trying to mention someone that isn't the one who invoked the command, I believe a mention in an embed is the best approach. If it was to mention the author, then using the "new" replies feature would've sounded nicer.

**Comment by jb3** [comment]
Closed in 2025 review of open issues. We do not feel there is a large need for this feature.
